src/styles/styled.js

```diff
@@ -20,28 +22,26 @@ const wrapper = StyledComponent =>
     );
   };
 
-const baseStyled = createStyled({wrapper, getInitialStyle, driver});
+type StyletronComponent<Props> = React.StatelessFunctionalComponent<Props> & {
+  // eslint-disable-next-line flowtype/no-weak-types
+  __STYLETRON__: any,
+};
+type WithTheme<Props> = {$theme: ThemeT} & Props;
+type StyleFn = {
+  (string, StyleObject): StyletronComponent<{}>,
 
-// TODO(#495): Need a flow expert to help remove this 'any' type
-// eslint-disable-next-line flowtype/no-weak-types
-export default function styledWrapper(...args: any) {
-  // If user is trying to style a styled component
-  // use withStyleDeep, otherwise use baseStyled
-  let styleFn = baseStyled;
-  if (args[0] && args[0].__STYLETRON__) {
-    styleFn = withStyleDeep;
-  }
-  // Also allow passing deep style overrides via $style prop
-  // Ex: <StyledDiv $style={{color: 'red'}} />
-  // Issue for supporting this natively in styletron:
-  // https://github.com/rtsao/styletron/issues/221
+  <Props>(string, (WithTheme<Props>) => StyleObject): StyletronComponent<Props>,
```

Comment:
Could we set here something as `<Props={}>` or maybe even `<Props=any>` (default `any` could turn this into non-breaking change?). So you don't ever have to do this:

```js
styled<{}>()
```

Reply:
i tried this out, but led to flow errors when combined with this function signature overload syntax. the angle brackets are [required](https://flow.org/en/docs/types/generics/#toc-adding-defaults-to-parameterized-generics) anyway, so i chose to not dig too deep on it